include/cbmqtm/cbmqtm_dbdistributor.php

```diff
@@ -69,14 +73,18 @@ private function insertMsg($channel, $producer, $consumer, $type, $share, $seque
 				'invalid' => 0,
 				'invalidreason' => '',
 				'userid' => $userid,
-				'information' => $information
+				'information' => $information,
+				'deliverStartTime' => $deliverRange['deliverStartTime'],
+				'deliverEndTime' => $deliverRange['deliverEndTime'],
+				'canSendOnSaturday' => $deliverRange['canSendOnSaturday'],
+				'canSendOnSunday' => $deliverRange['canSendOnSunday'],
 		));
 	}
 
 	public function getMessage($channel, $consumer, $producer = '*', $userid = '*') {
 		self::setDB();
-		$sql = 'select * from cb_messagequeue where deliverafter<=? and channel=? and consumer=?';
-		$params = array(date('Y-m-d H:i:s', time()), $channel, $consumer);
+		$sql = 'select * from cb_messagequeue where deliverafter<=? and channel=? and consumer=? and deliverStartTime<? and deliverEndTime>?';
```

Comment:
The query below is wrong because you are forcing the deliver range start and end which is not the case. Most messages on the queue will not have this restriction and can be delivered at any hour of the day, any day. The start-end, and day conditions must apply only to those messages that require it.

I think you have to do the same "if" code you did for Saturday/Sunday for the start-end

Reply:
i added something like this 
`
$sqlcheck = 'select deliverStartTime, deliverEndTime from cb_messagequeue';
$resultcheck = static::$db->pquery($sqlcheck);
if ($resultcheck) {
	$sql .= ' and deliverStartTime<? and deliverEndTime>?';
}
`